public/app/features/search/loaders.ts

```diff
@@ -39,19 +39,25 @@ export const loadFolderPage = (uid: string, activeChildId: string) => {
   return backendSrv.getFolderByUid(uid).then((folder) => {
     const folderTitle = folder.title;
     const folderUrl = folder.url;
+
     navModel.main.text = folderTitle;
 
-    const dashTab = navModel.main.children!.find((child: any) => child.id === 'manage-folder-dashboards');
+    const dashTab = navModel.main.children![0]!;
+    const permTab = navModel.main.children![1]!;
+    const settingsTab = navModel.main.children![2]!;
+
+    settingsTab!.url = folderUrl + '/settings';
     dashTab!.url = folderUrl;
+    permTab!.url = folderUrl + '/permissions';
+
+    navModel.main.children = [dashTab];
 
     if (folder.canAdmin) {
-      const permTab = navModel.main.children!.find((child: any) => child.id === 'manage-folder-permissions');
-      permTab!.url = folderUrl + '/permissions';
+      navModel.main.children.push(permTab);
+    }
 
-      const settingsTab = navModel.main.children!.find((child: any) => child.id === 'manage-folder-settings');
-      settingsTab!.url = folderUrl + '/settings';
-    } else {
-      navModel.main.children = [dashTab!];
+    if (folder.canEdit) {
```

Comment:
Think this should be `canSave`. Otherwise settings tab would be shown for viewers when viewers_can_edit is enabled. Ref: https://github.com/grafana/grafana/blob/85e186cf10b0d672be5ee17f906f19efb765889b/pkg/services/guardian/guardian.go#L48-L58

```suggestion
    if (folder.canSave) {
```